lib/filter/index.ts

```diff
@@ -24,8 +24,8 @@ const debug = newDebug('snyk:policy');
  * @param matchStrategy the strategy used to match vulnerabilities (defaults to 'packageManager')
  * @returns the filtered vulnerabilities object
  */
-function filter<T extends Vulnerability>(
-  vulns: VulnerabilityReport<T>,
+function filter<VulnType extends Vulnerability, ReportType>(
+  vulns: ReportType & VulnerabilityReport<VulnType>,
```

Comment:
The vulnerability object is mutated, so updating the vulns type and the return type to take this into account.